ui/src/components/CardDetailModal.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react';
+import Box from '@material-ui/core/Box';
+import Modal from '@material-ui/core/Modal';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { makeStyles } from '@material-ui/core/styles';
+
+import BaseballCard from './BaseballCard.jsx';
+import AuctionSessionDetail from './AuctionSessionDetail.jsx';
+import CardAuctionForm from './CardAuctionForm.jsx';
+
+const useStyles = makeStyles((theme) => {
+  return {
+    root: {
+      margin: 'auto', // theme.spacing(4),
+      marginTop: theme.spacing(12),
+      display: 'flex',
+      background: 'white',
+      borderRadius: '4px',
+      maxWidth: '960px',
+      alignItems: 'center',
+    },
+    card: {
+      width: '360px',
+    },
+    detail: {
+      padding: theme.spacing(3),
+      marginLeft: theme.spacing(2),
+      marginBottom: theme.spacing(2),
+      flex: 1,
+    },
+    loading: {
+      marginBottom: theme.spacing(2),
+    },
+    heading: {
+      fontWeight: 600,
+      fontSize: '20px',
+      marginBottom: theme.spacing(1),
+    },
+    bidForm: {
+      marginBottom: theme.spacing(1),
+    },
+    bidFormItem: {
+      marginBottom: theme.spacing(1),
+    },
+  };
+});
+
+const CardDetailModal = ({
+  open,
+  onClose,
+  playerName,
+  tokenPurses,
+  tokenPetname,
+  tokenDisplayInfo,
+  onGetCardDetail,
+  onBidCard,
+}) => {
+  const classes = useStyles();
+  const [state, setDetailState] = useState({});
+
+  useEffect(() => {
+    if (!playerName) {
+      setDetailState({
+        details: null,
+        error: null,
+      });
+      return;
+    }
+    onGetCardDetail(playerName)
+      .then((result) => {
```

Comment:
This is not likely a big deal, but this might be a race condition. If playerName changes before this completes, it's possible the details could be updated with the old playerName. One way around this is to use a cleanup function to cancel if playerName changes https://maxrozen.com/race-conditions-fetching-data-react-with-useeffect#useeffect-clean-up-function-with-boolean-flag